docs/source/developers/guide/tutorials/r_tutorial.rst

```diff
@@ -22,21 +22,429 @@
+
+And ``git diff`` to see the changes in the files in order to spot any error we might have made.
+
+.. code:: console
+
+   $ git diff
+   diff --git a/r/R/expression.R b/r/R/expression.R
+   index 37fc21c25..0e71803ec 100644
+   --- a/r/R/expression.R
+   +++ b/r/R/expression.R
+   @@ -70,6 +70,7 @@
+      "quarter" = "quarter",
+      # second is defined in dplyr-functions.R
+      # wday is defined in dplyr-functions.R
+   +  "mday" = "day",
+      "yday" = "day_of_year",
+      "year" = "year",
+
+   diff --git a/r/tests/testthat/test-dplyr-funcs-datetime.R b/r/tests/testthat/test-dplyr-funcs-datetime.R
+   index 359a5403a..228eca56a 100644
+   --- a/r/tests/testthat/test-dplyr-funcs-datetime.R
+   +++ b/r/tests/testthat/test-dplyr-funcs-datetime.R
+   @@ -444,6 +444,15 @@ test_that("extract wday from timestamp", {
+      )
+    })
+    
+   +test_that("extract mday from timestamp", {
+   +  compare_dplyr_binding(
+   +    .input %>%
+   +      mutate(x = mday(datetime)) %>%
+   +      collect(),
+   +    test_df
+   +  )
+   +})
+   +
+    test_that("extract yday from timestamp", {
+      compare_dplyr_binding(
+        .input %>%
+   @@ -626,6 +635,15 @@ test_that("extract wday from date", {
+      )
+    })
+
+   +test_that("extract mday from date", {
+   +  compare_dplyr_binding(
+   +    .input %>%
+   +      mutate(x = mday(date)) %>%
+   +      collect(),
+   +    test_df
+   +  )
+   +})
+   +
+    test_that("extract yday from date", {
+      compare_dplyr_binding(
+        .input %>%
+
+Everything looks OK. Now we can make the commit
+(save our changes to the branch history):
+
+.. code:: console
+
+   $ git commit -am "Adding a binding and a test for mday() lubridate"
+   [ARROW-14816 ed37d3a3b] Adding a binding and a test for mday() lubridate
+    2 files changed, 19 insertions(+)
+
+We can use ``git log`` to check the history of commits:
+
+.. code:: console
+
+   $ git log
+   commit ed37d3a3b3eef76b696532f10562fea85f809fab (HEAD -> ARROW-14816)
+   Author: Alenka Frim <frim.alenka@gmail.com>
+   Date:   Fri Jan 21 09:15:31 2022 +0100
+
+       Adding a binding and a test for mday() lubridate
+
+   commit c5358787ee8f7b80f067292f49e5f032854041b9 (upstream/master, upstream/HEAD, master, ARROW-15346, ARROW-10643)
+   Author: Krisztián Szűcs <szucs.krisztian@gmail.com>
+   Date:   Thu Jan 20 09:45:59 2022 +0900
+
+       ARROW-15372: [C++][Gandiva] Gandiva now depends on boost/crc.hpp which is missing from the trimmed boost archive
+       
+       See build error https://github.com/ursacomputing/crossbow/runs/4871392838?check_suite_focus=true#step:5:11762
+       
+       Closes #12190 from kszucs/ARROW-15372
+       
+       Authored-by: Krisztián Szűcs <szucs.krisztian@gmail.com>
+       Signed-off-by: Sutou Kouhei <kou@clear-code.com>
+
+If we started the branch some time ago, we may need to rebase
+to upstream master to make sure there are no merge conflicts:
+
+.. code:: console
+
+   $ git pull upstream master --rebase
```

Comment:
Should this be `git rebase` not `git pull`?

Reply:
If I understand correctly `git pull --rebase` is equal to `git fetch`& `git rebase`, just shorter (and there is something about saving you trouble when rebasing on a force-pushed rebase https://git-scm.com/book/en/v2/Git-Branching-Rebasing#_rebase_rebase_work)

Reply:
Oh awesome, learned something new today!